lib/workers/repository/update/branch/handle-existing.ts

```diff
@@ -34,14 +34,14 @@ export async function handlepr(config: BranchConfig, pr: Pr): Promise<void> {
         });
       }
     }
-    if (branchExists(config.branchName)) {
-      if (GlobalConfig.get('dryRun')) {
-        logger.info('DRY-RUN: Would delete branch ' + config.branchName);
-      } else {
-        await deleteBranch(config.branchName);
-      }
-    }
   } else if (pr.state === PrState.Merged) {
-    logger.debug({ pr: pr.number }, 'Merged PR is blocking this branch');
+    logger.debug({ pr: pr.number }, 'Merged PR found');
+  }
+  if (branchExists(config.branchName)) {
+    if (GlobalConfig.get('dryRun')) {
+      logger.info('DRY-RUN: Would delete branch ' + config.branchName);
+    } else {
+      await deleteBranch(config.branchName);
```

Comment:
Would this end up deleting *itself* in some cases? e.g.

- Branch/PR created
- PR merged, branch remains
- Another PR is required, so branch is deleted per the above
- New branch is created next run, but not PR yet (because prCreation=not-pending)
- Next run, branch is deleted again

Reply:
Branch/PR created - No
PR merged, branch remains - Yes
Another PR is required, so branch is deleted per the above - Yes
New branch is created next run, but not PR yet (because prCreation=not-pending) - No
Next run, branch is deleted again - No

The condition depends on PR only no relation with branch. If Pr is open then we don't do anything but if it's closed/merged we delete the branch. That said I should have been more cautious with the placement of the block.